### DISCUSSION TITLE
Release 21.08

### DISCUSSION THREAD

**Discussion by svpcom** [svpcom/wfb-ng#163] [General]
This release is **NOT** compatible with previous releases due to packet header change. You need to upgrade both drone and GS sides.

Changes:
-------------
1. Fix bug with FEC if first packet of block was not lost
2. Add TX FEC timeout option: emit empty packets if block is open but no input.
    This can reduce video latency and allow to use K/N FEC for telemetry where K > 1
    This option is disabled by default, use it on your own risk.
3. Add serial endpoint support. You can use `peer = 'serial:ttyX:1500000` to connect directly to autopilot without mavlink-router
4. Add ARM/DISARM handler. You can use `call_on_arm = '/some/binary'` and/or `call_on_disarm = '/other/binary'` in telemetry sections of drone and GS. For example this can be used for trigger video recording.
5. Add telemetry mirroring on GS. You can use it to mirror all incoming mavlink traffic to some UDP port (for example to OSD app) without mavlink-router
6. Python3 is now default. Python2 support will be dropped in the next release

<hr /><em>This discussion was created from the release <a href='https://github.com/svpcom/wifibroadcast/releases/tag/wifibroadcast-21.08'>Release 21.08</a>.</em>

**Comment by IGNNE** [comment]
Seems to work well, on the RasPi side I needed to link libsodium.so.23 to/as libsodium.so.13 (Raspbian oldstable).
